services/streammanager/kafka/client/producer.go

```diff
@@ -117,5 +126,22 @@ func (p *Producer) Publish(ctx context.Context, msgs ...Message) error {
 			Headers: headers,
 		}
 	}
+
 	return p.writer.WriteMessages(ctx, messages...)
 }
+
+var tempError interface{ Temporary() bool }
+
+func IsProducerErrTemporary(err error) bool {
+	isTransientNetworkError := errors.Is(err, io.ErrUnexpectedEOF) ||
+		errors.Is(err, syscall.ECONNREFUSED) ||
+		errors.Is(err, syscall.ECONNRESET) ||
+		errors.Is(err, syscall.EPIPE)
+	if isTransientNetworkError {
+		return true
+	}
+	if errors.As(err, &tempError) {
+		return tempError.Temporary()
+	}
+	return false
```

Comment:
I think the alternative is to treat all errors as retryable, even errors where a message really can't be processed, for example if you push a 20MB message and the `message.max.bytes` on Kafka was lower. This would be a new behaviour that could add sensible stress to the system in certain situations. Consider that we retry messages for up to 3 hours before dropping them so it could also make us susceptible to attacks. In my opinion it is best to stick to the Kafka documentation and treat only the errors that they say are temporary as such.